plugins/modules/openshift_route.py

```diff
@@ -329,6 +338,10 @@ def __init__(self):
         self.module = AnsibleModule(
             argument_spec=self.argspec,
             supports_check_mode=True,
+            required_if=[
+                ('state', 'present', ['service']),
+                ('state', 'absent', ['name'])
```

Comment:
This is a breaking change since you can currently delete a route by specifying only the service name. I'm not sure there's a compelling reason to use this restriction. If we want to, though, we need to deprecate the behavior, first.

Reply:
updated